controller/api/destination/server.go

```diff
@@ -38,185 +41,141 @@ type server struct {
-						Pod:  &pod,
-					},
-				)
-			}
-
-			discoverySP.PortEndpoints[port] = &podAddrs
-		}
-
-		s.log.Debugf("ServicePorts[%s]: %+v", serviceID, discoverySP)
-		rsp.ServicePorts[serviceID.String()] = &discoverySP
+func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetProfileServer) error {
+	log := s.log
+	client, _ := peer.FromContext(stream.Context())
+	if client != nil {
+		log = log.WithField("remote", client.Addr)
 	}
+	log.Debugf("GetProfile(%+v)", dest)
 
-	return &rsp, nil
-}
+	translator := newProfileTranslator(stream, log)
 
-func (s *server) streamResolution(host string, port int, stream pb.Destination_GetServer) error {
-	listener := newEndpointListener(stream, s.k8sAPI.GetOwnerKindAndName, s.enableH2Upgrade, s.controllerNS, s.identityTrustDomain)
+	primary, secondary := newFallbackProfileListener(translator)
 
-	resolverCanResolve, err := s.resolver.canResolve(host, port)
-	if err != nil {
-		return fmt.Errorf("resolver [%+v] found error resolving host [%s] port [%d]: %v", s.resolver, host, port, err)
-	}
-	if !resolverCanResolve {
-		return fmt.Errorf("cannot find resolver for host [%s] port [%d]", host, port)
+	if dest.GetContextToken() != "" {
+		s.profiles.Subscribe(dest.GetPath(), dest.GetContextToken(), primary)
+		defer s.profiles.Unsubscribe(dest.GetPath(), dest.GetContextToken(), primary)
 	}
-	return s.resolver.streamResolution(host, port, listener)
-}
 
-func getHostAndPort(dest *pb.GetDestination) (string, int, error) {
-	if dest.Scheme != "k8s" {
-		err := fmt.Errorf("Unsupported scheme %s", dest.Scheme)
-		log.Error(err)
-		return "", 0, err
-	}
-	hostPort := strings.Split(dest.Path, ":")
-	if len(hostPort) > 2 {
-		err := fmt.Errorf("Invalid destination %s", dest.Path)
-		log.Error(err)
-		return "", 0, err
-	}
-	host := hostPort[0]
-	port := 80
-	if len(hostPort) == 2 {
-		var err error
-		port, err = strconv.Atoi(hostPort[1])
-		if err != nil {
-			err = fmt.Errorf("Invalid port %s", hostPort[1])
-			log.Error(err)
-			return "", 0, err
-		}
-	}
-	return host, port, nil
-}
+	s.profiles.Subscribe(dest.GetPath(), "", secondary)
+	defer s.profiles.Unsubscribe(dest.GetPath(), "", secondary)
 
-func buildResolver(
-	k8sDNSZone string,
-	k8sAPI *k8s.API,
-) (streamingDestinationResolver, error) {
-	k8sDNSZoneLabels := []string{}
-	if k8sDNSZone != "" {
-		var err error
-		k8sDNSZoneLabels, err = splitDNSName(k8sDNSZone)
-		if err != nil {
-			return nil, err
-		}
+	select {
+	case <-s.shutdown:
+	case <-stream.Context().Done():
+		log.Debugf("GetProfile(%+v) cancelled", dest)
 	}
 
-	k8sResolver := newK8sResolver(k8sDNSZoneLabels, newEndpointsWatcher(k8sAPI), newProfileWatcher(k8sAPI))
-
-	log.Infof("Built k8s name resolver")
+	return nil
+}
 
-	return k8sResolver, nil
+func (s *server) Endpoints(ctx context.Context, params *discoveryPb.EndpointsParams) (*discoveryPb.EndpointsResponse, error) {
+	s.log.Debugf("serving endpoints request")
+
+	// servicePorts := e.getState()
```

Comment:
Did you mean to leave the commented-out section in this file?

Reply:
Thanks for bringing this up.  The plan is to remove this implementation while we figure out the right long term solution.  I'll remove the commented out code.